lib/perimeterx/internal/validators/perimeter_x_cookie_validator.rb

```diff
@@ -18,11 +18,19 @@ def initialize(px_config)
 
     def verify(px_ctx)
       begin
-        # Case no cookie
-        if px_ctx.context[:px_cookie].empty?
-          @logger.warn("PerimeterxCookieValidator:[verify]: cookie not found")
+        # Error cases
```

Comment:
for error cases
the check should be if we're in mobile scenario
psedo:
```if cookie origin
    if 1
    else if 2
    else if 3
    end if
endif
```

Reply:
OK, agreed. In such a case I will separate from the generic `px_ctx.context[:px_cookie].empty?` check (such that when `px_ctx.context[:px_cookie] == "1"`, it will also result in `PxModule::NO_COOKIE`).

Reply:
Also adding [EN-4509](https://perimeterx.atlassian.net/browse/EN-4509) (empty token should return `cookie_decryption_failed`)